util/pace/util/grid/generation.py

```diff
@@ -1988,6 +2173,31 @@ def _calculate_center_vectors(self):
         self._ec2_64 = ec2_64
         return ec1, ec2
 
+    def _calculate_center_vectors_cartesian(self):
+        ec1_64 = self.quantity_factory.zeros(
+            [util.X_DIM, util.Y_DIM, self.CARTESIAN_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        ec2_64 = self.quantity_factory.zeros(
+            [util.X_DIM, util.Y_DIM, self.CARTESIAN_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        ec1_64.data[:, :, 0] = 1.0
+        ec1_64.data[:, :, 1:3] = 0.0
+        ec2_64.data[:, :, 0] = 0.0
+        ec2_64.data[:, :, 1] = 1.0
+        ec2_64.data[:, :, 2] = 0.0
```

Comment:
if ec1 and ec2 are initialized as 0 I think you only need to set `ec1_64.data[:, :, 0] = 1.0` and `ec2_64.data[:, :, 1] = 1.0`:

```suggestion
        ec1_64.data[:, :, 0] = 1.0
        ec2_64.data[:, :, 1] = 1.0
```